### ISSUE TITLE
Gradient checkpointing 

### ISSUE THREAD

**Issue by lakshya-4gp** [google-research/tuning_playbook#45]
I believe the gradient checkpointing can be very useful if you have to maintain some minimum batch size, and you can't do that with your hardware. I was training a Nerf model, and i had to use it to atleast maintain some minimum number of samples in a batch. Any thoughts?

**Comment by znado** [comment]
Do you mean gradient accumulation?

I agree that if you batch size is absolutely minuscule (<4?), then _maybe_ that could improve performance. Although I would want to see some tuned side-by-side studies showing that is the case.
